Cargo.toml

```diff
@@ -16,11 +16,25 @@ prost = "0.11.6"
 reqwest = { version = "0.11.13", features = ["rustls-tls"] }
 tokio = { version = "1", default-features = false, features = ["time"] }
 rand = "0.8.5"
+bitcoin = "0.31.1"
+tokio-tungstenite = "0.21.0"
+tungstenite = "0.21.0"
+async-trait = "0.1.77"
+futures = "0.3.30"
+url = "2.5.0"
```

Comment:
We also use this library to handle parsing the domain name, parsing the query parameters, and adding additional `sig` and `key` query parameters. We can implement this functionality, but it would be nice to use the library here.